level00/CoffeeMachineTests.hs

```diff
@@ -0,0 +1,105 @@
+{-# LANGUAGE RankNTypes #-}
+module CoffeeMachineTests (propertyTests) where
+
+import Control.Lens (Prism', prism, matching, preview, review)
+
+import Test.Tasty (TestTree, testGroup)
+import Test.Tasty.Hedgehog (testProperty)
+
+import Hedgehog
+import qualified Hedgehog.Gen as Gen
+import qualified Hedgehog.Range as Range
+
+-- Starting
+addTen :: Int -> Int
+addTen = (+10)
+
+prop_addTen :: Property
+prop_addTen = property $ do
+  n <- forAll $ Gen.int (Range.linear 0 1000)
+  addTen n === n + 10
+
+badReverse :: [a] -> [a]
+badReverse [] = []
+badReverse (_:xs) = reverse xs
+
+prop_badReverse :: Property
+prop_badReverse = property $ do
+  xs <- forAll (Gen.list (Range.linear 0 1000) Gen.bool)
+  badReverse (badReverse xs) === xs
+-- End Starting
+
+-- Need a middle ground exercise
```

Comment:
What about some of the properties from that recent talk which introduced groups?